### PR TITLE
Add python 3 support

### PR DESCRIPTION
**Pull request by michaelbausor**
This PR:
- Adds support for python 3
- Adds support for CI
- Update baselines for latest protoc

### CODE REVIEW

File: gapic_plugin.py [hunk 1]
```diff
@@ -157,10 +157,17 @@ def main(data):
   return output
 
 if __name__ == '__main__':
+  try:
```
Comment (lukesneeringer) [inline gapic_plugin.py]: I am curious why this change was necessary. `sys.stdin.read()` and `sys.stdout.write()` are valid in Python 3.

Comment (michaelbausor) [inline gapic_plugin.py]: This is necessary to work with binary data in Python 3 - see the note here: https://docs.python.org/3/library/sys.html#sys.stdin
